src/pj_obs_api.c

```diff
@@ -296,10 +315,44 @@ size_t proj_transform (
         *z = coord.xyzt.z;
     if (nt==1)
         *t = coord.xyzt.t;
-    
+
     return i;
 }
 
+/*****************************************************************************/
+int proj_transform_obs (PJ *P, enum proj_direction direction, size_t n, PJ_OBS *obs) {
+/******************************************************************************
+    Batch transform an array of PJ_OBS.
+
+    Returns 0 if all observations are transformed without error, otherwise
+    returns error number.
```

Comment:
Nowadays, wouldn't returning NaN be a better indication of a problem?  HUGE_VAL is a reasonable valid result (a point that transforms to infinity).  Also making sure that all the transformation routines deal with arguments which are NaN would be nice.